precompiles/ArbAggregator.go

```diff
@@ -20,87 +16,111 @@ type ArbAggregator struct {
+// Get the addresses of all current batch posters
+func (con ArbAggregator) GetBatchPosters(c ctx, evm mech) ([]addr, error) {
+	return c.State.L1PricingState().BatchPosterTable().AllPosters()
 }
 
-// Sets the chain's default aggregator (caller must be the current default aggregator, its fee collector, or an owner)
+// [Deprecated]
 func (con ArbAggregator) SetDefaultAggregator(c ctx, evm mech, newDefault addr) error {
-	l1State := c.State.L1PricingState()
-	defaultAgg, err := l1State.DefaultAggregator()
+	isOwner, err := c.State.ChainOwners().IsMember(c.caller)
 	if err != nil {
 		return err
 	}
-	allowed, err := accountIsAggregatorOrCollectorOrOwner(c.caller, defaultAgg, c.State)
+	if !isOwner {
+		return ErrNotOwner
+	}
+	batchPosterTable := c.State.L1PricingState().BatchPosterTable()
+	isBatchPoster, err := batchPosterTable.ContainsPoster(newDefault)
 	if err != nil {
 		return err
 	}
-	if !allowed {
-		return errors.New("Only the current default (or its fee collector / chain owner) may change the default")
+	if !isBatchPoster {
+		_, err = batchPosterTable.AddPoster(newDefault, newDefault)
+		if err != nil {
+			return err
+		}
 	}
-	return l1State.SetDefaultAggregator(newDefault)
-}
-
-// Get the aggregator's compression ratio, measured in basis points
-func (con ArbAggregator) GetCompressionRatio(c ctx, evm mech, aggregator addr) (uint64, error) {
-	ratio, err := c.State.L1PricingState().AggregatorCompressionRatio(aggregator)
-	return uint64(ratio), err
+	return nil
 }
 
-// Set the aggregator's compression ratio, measured in basis points
-func (con ArbAggregator) SetCompressionRatio(c ctx, evm mech, aggregator addr, newRatio uint64) error {
-	allowed, err := accountIsAggregatorOrCollectorOrOwner(c.caller, aggregator, c.State)
+func (con ArbAggregator) AddBatchPoster(c ctx, evm mech, newBatchPoster addr) error {
+	isOwner, err := c.State.ChainOwners().IsMember(c.caller)
 	if err != nil {
 		return err
 	}
-	if !allowed {
-		return errors.New("Only an aggregator (or its fee collector / chain owner) may change its compression ratio")
+	if !isOwner {
+		return ErrNotOwner
 	}
-	return c.State.L1PricingState().SetAggregatorCompressionRatio(aggregator, arbmath.Bips(newRatio))
+	batchPosterTable := c.State.L1PricingState().BatchPosterTable()
+	isBatchPoster, err := batchPosterTable.ContainsPoster(newBatchPoster)
+	if err != nil {
+		return err
+	}
+	if !isBatchPoster {
+		_, err = batchPosterTable.AddPoster(newBatchPoster, newBatchPoster)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
 }
 
-// Gets an aggregator's fee collector
-func (con ArbAggregator) GetFeeCollector(c ctx, evm mech, aggregator addr) (addr, error) {
-	return c.State.L1PricingState().AggregatorFeeCollector(aggregator)
+// Gets a batch poster's fee collector
+func (con ArbAggregator) GetFeeCollector(c ctx, evm mech, batchPoster addr) (addr, error) {
+	posterInfo, err := c.State.L1PricingState().BatchPosterTable().OpenPoster(batchPoster)
```

Comment:
I don't think this should add the poster to the table if it don't exist, as it currently does. We should probably add an argument to OpenPoster for if it should error if the poster doesn't exist.

Reply:
Fixed